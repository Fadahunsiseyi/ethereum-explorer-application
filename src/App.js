import { Routes, Route } from 'react-router-dom';
import GetApi from './api/Api';
import Navbar from './components/Navbar';
import HomePage from './pages/HomePage';
import NotFound from './pages/NotFound';
import TransactionPage from './pages/TransactionPage';

GetApi();

function App() {
  return (
    <div className="App">
      <Navbar />
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/transaction" element={<TransactionPage />} />
        <Route path="*" element={<NotFound />} />
      </Routes>
    </div>
  );
}

export default App;
