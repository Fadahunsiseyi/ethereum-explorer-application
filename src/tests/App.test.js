import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import App from '../App';

afterEach(() => cleanup());

describe('Test App.js', () => {
  it('Test if the page loads', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
