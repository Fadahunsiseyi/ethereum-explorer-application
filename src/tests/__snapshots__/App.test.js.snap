// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App.js Test if the page loads 1`] = `
<div
  className="App"
>
  <nav>
    <button
      className="chevBtn"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="chevIcon"
        fill="none"
        stroke="currentColor"
        strokeWidth={2}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 19l-7-7 7-7"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </button>
    <div
      className="search-input"
    >
      <input
        id="search-input"
        name="search input"
        onClick={[Function]}
        placeholder="Search Transaction..."
      />
    </div>
    <div
      className="icons"
    >
      <svg
        className="cogIcon"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M11.49 3.17c-.38-1.56-2.6-1.56-2.98 0a1.532 1.532 0 01-2.286.948c-1.372-.836-2.942.734-2.106 2.106.54.886.061 2.042-.947 2.287-1.561.379-1.561 2.6 0 2.978a1.532 1.532 0 01.947 2.287c-.836 1.372.734 2.942 2.106 2.106a1.532 1.532 0 012.287.947c.379 1.561 2.6 1.561 2.978 0a1.533 1.533 0 012.287-.947c1.372.836 2.942-.734 2.106-2.106a1.533 1.533 0 01.947-2.287c1.561-.379 1.561-2.6 0-2.978a1.532 1.532 0 01-.947-2.287c.836-1.372-.734-2.942-2.106-2.106a1.532 1.532 0 01-2.287-.947zM10 13a3 3 0 100-6 3 3 0 000 6z"
          fillRule="evenodd"
        />
      </svg>
      <svg
        className="micIcon"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M7 4a3 3 0 016 0v4a3 3 0 11-6 0V4zm4 10.93A7.001 7.001 0 0017 8a1 1 0 10-2 0A5 5 0 015 8a1 1 0 00-2 0 7.001 7.001 0 006 6.93V17H6a1 1 0 100 2h8a1 1 0 100-2h-3v-2.07z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </nav>
  <div
    className="hero-container"
  >
    <svg
      preserveAspectRatio="xMidYMid"
      version="1.1"
      viewBox="0 0 256 417"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <polygon
          fill="#343434"
          points="127.9611 0 125.1661 9.5 125.1661 285.168 127.9611 287.958 255.9231 212.32"
        />
        <polygon
          fill="#8C8C8C"
          points="127.962 0 0 212.32 127.962 287.959 127.962 154.158"
        />
        <polygon
          fill="#3C3C3B"
          points="127.9611 312.1866 126.3861 314.1066 126.3861 412.3056 127.9611 416.9066 255.9991 236.5866"
        />
        <polygon
          fill="#8C8C8C"
          points="127.962 416.9052 127.962 312.1852 0 236.5852"
        />
        <polygon
          fill="#141414"
          points="127.9611 287.9577 255.9211 212.3207 127.9611 154.1587"
        />
        <polygon
          fill="#393939"
          points="0.0009 212.3208 127.9609 287.9578 127.9609 154.1588"
        />
      </g>
    </svg>
    <div
      className="title-wrapper"
    >
      <h2 />
      <h6 />
    </div>
  </div>
  <div
    className="order-container"
  >
    <h5>
      ORDER BY
    </h5>
    <div
      className="select-input css-b62m3t-container"
      onKeyDown={[Function]}
    >
      <span
        className="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        className="css-1f43avz-a11yText-A11yText"
      />
      <div
        className=" css-1s2u09g-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-319lph-ValueContainer"
        >
          <div
            className=" css-14el2xx-placeholder"
            id="react-select-2-placeholder"
          >
            Select Order
          </div>
          <div
            className=" css-6j8wv5-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-2-placeholder"
              aria-expanded={false}
              aria-haspopup={true}
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              className=""
              disabled={false}
              id="react-select-2-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              role="combobox"
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "color": "inherit",
                  "font": "inherit",
                  "gridArea": "1 / 2",
                  "label": "input",
                  "margin": 0,
                  "minWidth": "2px",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "100%",
                }
              }
              tabIndex={0}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className=" css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className=" css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
      <input
        name="Select"
        type="hidden"
        value=""
      />
    </div>
  </div>
  <ul
    className="transactions-container"
  />
</div>
`;
